src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1851,10 +1853,21 @@ void parser::parse_and_visit_switch(parse_visitor_base &v) {
         });
         this->skip();
       } else {
-        this->parse_and_visit_expression(
+        expression *ast = this->parse_expression(
             v, precedence{
                    .colon_type_annotation = allow_type_annotations::never,
                });
+
+        source_code_span expression_case_span = ast->span();
+        auto [_it, inserted] = cases.insert(expression_case_span.string_view());
+        if (!inserted) {
+          this->diag_reporter_->report(
+              diag_duplicated_cases_in_switch_statement{
+                  .first_switch_case =
+                      source_code_span(_it->begin(), _it->end()),
```

Comment:
`begin` and `end` return iterators, but `source_code_span` expects pointers. This is causing build failures with MSVC. You should use `_it->data()` and `_it->data() + _it->size()` instead.

Reply:
Thank you, I didn't realized that the constructor expected pointers, now I was just looking to the `source_code_span` class and saw that there's a `source_code_span::unit` member function, since we are using this span only to point to the original `case` do you think it would be better to use that function like?
`.first_switch_case = source_code_span::unit(_it->data())`

Or should I just use the constructor

Reply:
No, you don't want to use `unit`. `unit` is for diagnostics which point between two characters. Use the two-pointer constructor.